sphinx_qt_documentation/__init__.py

```diff
@@ -21,14 +21,26 @@
 from typing import List, Optional, Dict, Any
 from sphinx.locale import get_translation
 from sphinx.ext.intersphinx import InventoryAdapter
-from qtpy.QtCore import Signal
 
 _ = get_translation("sphinx")
 
-try:
-    from qtpy import QT_VERSION
-except ImportError:
-    QT_VERSION = None
+
+def _get_signal_and_version():
+    version_mapping = {'qtpy': lambda: 'QT_VERSION', 'Qt': lambda: '__qt_version__',
+                       'Pyside2': lambda: importlib.import_module('PySide2.QtCore').__version__,
+                       'PyQt5': lambda: importlib.import_module('PyQt5.QtCore').QT_VERSION_STR}
+    for module_name in version_mapping.keys():
+        try:
+            core = importlib.import_module(f'{module_name}.QtCore')
+            signal = getattr(core, 'Signal')
+            return signal, version_mapping[module_name]()
+        except (ModuleNotFoundError, ImportError):
+            continue
+    # might be better to mock a SSignal class here
+    return None, None
```

Comment:
In this moment I do not remember why I let `QT_VERSION` to  `None`, but second argument of isinstance cannot be `None` https://github.com/Czaki/sphinx-qt-documentation/blob/84b915442b5aa6054a8c28341e3ce3009dbebdf9/sphinx_qt_documentation/__init__.py#L165

I prefer to raise exception in this case.

Reply:
Right. I'll raise a RuntimeError instead.